mantle/kola/tests/misc/boot-mirror.go

```diff
@@ -0,0 +1,427 @@
+				  ]
+				},
+				{
+				  "devices": [
+					"/dev/disk/by-partlabel/root-1",
+					"/dev/disk/by-partlabel/root-2"
+				  ],
+				  "level": "raid1",
+				  "name": "md-root"
+				}
+			  ]
+			}
+		  }`)
+)
+
+func init() {
+	register.RegisterTest(&register.Test{
+		Run:              runBootMirrorTest,
+		ClusterSize:      0,
+		Name:             `coreos.boot-mirror`,
+		Platforms:        []string{"qemu-unpriv"},
+		ExcludeFirmwares: []string{"uefi"},
+		Tags:             []string{"boot-mirror", "raid1"},
+	})
+	register.RegisterTest(&register.Test{
+		Run:         runBootMirrorLUKSTest,
+		ClusterSize: 0,
+		Name:        `coreos.boot-mirror.luks`,
+		Platforms:   []string{"qemu-unpriv"},
+		// aarch64 and ppc64le are added temporarily to avoid test failure
+		// until we support specifying FCC directly in kola.
+		ExcludeArchitectures: []string{"aarch64", "ppc64le", "s390x"}, // no TPM support for s390x in qemu
+		ExcludeFirmwares:     []string{"uefi"},
+		Tags:                 []string{"boot-mirror", "luks", "raid1", "tpm2", kola.NeedsInternetTag},
+	})
+}
+
+// runBootMirrorTest verifies if the boot-mirror RAID1
+// flow works properly in both BIOS and UEFI modes.
+func runBootMirrorTest(c cluster.TestCluster) {
+	var m platform.Machine
+	var err error
+	options := platform.QemuMachineOptions{
+		MachineOptions: platform.MachineOptions{
+			AdditionalDisks: []string{"5120M", "5120M"},
+			MinMemory:       4096,
+		},
+	}
+	m, err = c.Cluster.(*unprivqemu.Cluster).NewMachineWithQemuOptions(bootmirror, options)
+	if err != nil {
+		c.Fatal(err)
+	}
+	// Check for root
+	checkIfMountpointIsRaid(c, m, "/sysroot")
+	fsTypeForRoot := c.MustSSH(m, "findmnt -nvr /sysroot -o FSTYPE")
+	if strings.Compare(string(fsTypeForRoot), "xfs") != 0 {
+		c.Fatalf("didn't match fstype for root")
+	}
+	bootMirrorSanityTest(c, m)
+
+	detachPrimaryBlockDevice(c, m)
+
+	verifyBootMirrorAfterReboot(c, m)
+}
+
+// runBootMirrorLUKSTest verifies if the boot-mirror+LUKS RAID1
+// flow works properly in both BIOS and UEFI modes.
+func runBootMirrorLUKSTest(c cluster.TestCluster) {
+	var m platform.Machine
+	var err error
+	options := platform.QemuMachineOptions{
+		MachineOptions: platform.MachineOptions{
+			AdditionalDisks: []string{"5120M"},
+			MinMemory:       4096,
+		},
+	}
+	m, err = c.Cluster.(*unprivqemu.Cluster).NewMachineWithQemuOptions(bootmirrorluks, options)
+	if err != nil {
+		c.Fatal(err)
+	}
+	rootOutput := c.MustSSH(m, "sudo mdadm --export --detail /dev/md/md-root")
+	if !strings.Contains(string(rootOutput), "/dev/vda4") || !strings.Contains(string(rootOutput), "/dev/vdb4") {
+		c.Fatalf("root raid device missing; expected devices: %v", string(rootOutput))
+	}
+	bootOutput := c.MustSSH(m, "sudo mdadm --export --detail /dev/md/md-boot")
+	if !strings.Contains(string(bootOutput), "/dev/vda3") || !strings.Contains(string(bootOutput), "/dev/vdb3") {
+		c.Fatalf("root raid device missing; expected devices: %v", string(rootOutput))
+	}
```

Comment:
Any reason these can't be moved into `bootMirrorSanityTest` so they're both inside a named run block as well as run on both LUKS & non-LUKS? (Same with below checks but into `verifyBootMirrorAfterReboot`)

Reply:
> Any reason these can't be moved into `bootMirrorSanityTest` so they're both inside a named run block as well as run on both LUKS & non-LUKS? (Same with below checks but into `verifyBootMirrorAfterReboot`)

Actually, both tests are using a different number of devices, so the conditional statement would differ in each case.
1. `coreos.boot-mirror`- 3 (removing `/dev/sdc(3/4)`)
2. `coreos.boot-mirror.luks`- 2 (removing `/dev/sdb(3/4)`)